tests/test_compute_utils.py

```diff
@@ -572,14 +577,145 @@ def test_surface_min_max(self):
+        """Test vectorized computation of cubic polynomial exact roots."""
+        cubic = 4
+        poly = np.arange(-60, 60).reshape(cubic, 6, -1)
+        poly[0] = np.where(poly[0] == 0, np.ones_like(poly[0]), poly[0])
+        poly = poly * np.e * np.pi
+        assert np.unique(poly.shape).size == poly.ndim
+        shift = np.arange(10)
+        assert np.unique(poly.shape + shift.shape).size == poly.ndim + shift.ndim
+        out = cubic_poly_roots(poly, shift, real=False)
+        for j in range(poly.shape[1]):
+            for k in range(poly.shape[2]):
+                for s in range(shift.size):
+                    a, b, c, d = poly[:, j, k]
+                    d = d - shift[s]
+                    root_finds = np.sort_complex(np.roots([a, b, c, d]))
+                    np.testing.assert_allclose(out[s, j, k], root_finds)
+
+    @pytest.mark.unit
+    def test_polyint(self):
+        """Test vectorized computation of polynomial primitive."""
+        quintic = 6
+        poly = np.arange(-90, 90).reshape(quintic, 3, -1) * np.e * np.pi
+        assert np.unique(poly.shape).size == poly.ndim
+        out = polyint(poly)
+        for j in range(poly.shape[1]):
+            for k in range(poly.shape[2]):
+                np.testing.assert_allclose(out[:, j, k], np.polyint(poly[:, j, k])[:-1])
+
+    @pytest.mark.unit
+    def test_polyder(self):
+        """Test vectorized computation of polynomial derivative."""
+        quintic = 6
+        poly = np.arange(-90, 90).reshape(quintic, 3, -1) * np.e * np.pi
+        assert np.unique(poly.shape).size == poly.ndim
+        out = polyder(poly)
+        for j in range(poly.shape[1]):
+            for k in range(poly.shape[2]):
+                np.testing.assert_allclose(out[:, j, k], np.polyder(poly[:, j, k]))
 
-@pytest.mark.unit
-def test_rotation_matrix():
-    """Test that rotation_matrix works with fwd & rev AD for axis=[0, 0, 0]."""
-    dfdx_fwd = jax.jacfwd(rotation_matrix)
-    dfdx_rev = jax.jacrev(rotation_matrix)
-    x0 = jnp.array([0.0, 0.0, 0.0])
+    @pytest.mark.unit
+    def test_polyeval(self):
+        """Test vectorized computation of polynomial evaluation."""
+        quintic = 6
+        poly = np.arange(-90, 90).reshape(quintic, 3, -1) * np.e * np.pi
+        assert np.unique(poly.shape).size == poly.ndim
+        x = np.linspace(0, 20, poly.shape[1] * poly.shape[2]).reshape(
+            poly.shape[1], poly.shape[2]
+        )
+        x = np.stack([x, x * 2], axis=-1)
+        x = np.stack([x, x * 2, x * 3, x * 4], axis=-1)
+        assert np.unique(x.shape).size == x.ndim
+        assert poly.shape[1:] == x.shape[: poly.ndim - 1]
+        assert np.unique((poly.shape[0],) + x.shape[poly.ndim - 1 :]).size == x.ndim - 1
+        out = polyeval(poly, x)
+        for j in range(poly.shape[1]):
+            for k in range(poly.shape[2]):
+                np.testing.assert_allclose(out[j, k], np.poly1d(poly[:, j, k])(x[j, k]))
+
+    # TODO: FIXME
```

Comment:
I'll make this root finding version a regression test if the spline one is confirmed to work